src/core/core.scale.js

```diff
@@ -90,10 +79,108 @@ function getPixelForGridLine(scale, index, offsetGridLines) {
 	return lineValue;
 }
 
-function computeTextSize(context, tick, font) {
-	return helpers.isArray(tick) ?
-		helpers.longestText(context, font, tick) :
-		context.measureText(tick).width;
+function measureText(ctx, data, gc, string) {
+	var key = ctx.font + ';' + string;
+	var textWidth = data[key];
+
+	if (!textWidth) {
+		textWidth = data[key] = ctx.measureText(string).width;
+		gc.push(key);
+	}
+	return textWidth;
+}
+
+function computeLabelSizes(ctx, tickFonts, ticks, cache) {
+	var length = ticks.length;
+	var widths = [];
+	var heights = [];
+	var data, gc, gcLen, i, j, jlen, tick, label, tickFont, width, height, nestedLabel, widest, highest;
+
+	cache = cache || {};
```

Comment:
I think writing this as the following would be more consistent with how I normally see this done in Chart.js:
```
	var gcLen, i, j, jlen, tick, label, tickFont, width, height, nestedLabel, widest, highest;

	var data = cache.data = cache.data || {};
	var gc = cache.garbageCollect = cache.garbageCollect || [];
	cache = cache || {};
`

Reply:
If `cache` is undefined, `cache.data` causes an error, so this order cannot be changed. However, `scale.longestTextCache` cannot be undefined by this code https://github.com/chartjs/Chart.js/blob/a984162c7e8492b22fa857bf701a4f2adfffd457/src/core/core.scale.js#L238 `cache = cache || {};` is probably not needed.

Reply:
Nevermind. I didn't notice `cache` was used by those later variables